tests/system_tests/conv1d/conv1d.py

```diff
@@ -0,0 +1,186 @@
+    binfile_dir = "./tests/system_tests/conv1d/build_dir_output"
+    binfile_path = Path(binfile_dir + "/output/env5_top_reconfig.bin")
+    skeleton_id = list(range(15, -1, -1))
+    skeleton_id_as_bytearray = bytearray()
+    for x in skeleton_id:
+        skeleton_id_as_bytearray.extend(
+            x.to_bytes(length=1, byteorder="little", signed=False)
+        )
+
+    total_bits = 8
+    frac_bits = 2
+    num_inputs = 5
+    kernel_size = 3
+    num_in_channels = 1
+    num_out_channels = 1
+    # batches = 2
+    num_outputs = num_inputs - kernel_size + 1
+    nn = create_vhd_files(
+        output_dir=vhdl_dir,
+        num_inputs=num_inputs,
+        num_outputs=num_outputs,
+        num_in_channels=num_in_channels,
+        num_out_channels=num_out_channels,
+        kernel_size=kernel_size,
+        total_bits=total_bits,
+        frac_bits=frac_bits,
+        skeleton_id=skeleton_id,
+    )
+
+    fxp_conf = FixedPointConfig(total_bits, frac_bits)
+
+    # inputs = fxp_conf.as_rational(fxp_conf.as_integer(torch.rand(batches, num_in_channels, num_inputs)))
+    inputs = fxp_conf.as_rational(
+        fxp_conf.as_integer(
+            torch.Tensor(
+                [
+                    [[0.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 1.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 1.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 1.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 0.0, 1.0]],
+                    [[2.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 1.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 1.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 1.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 0.0, 1.0]],
+                    [[3.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 2.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 2.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 2.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 0.0, 2.0]],
+                    [[1.0, 1.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 1.0, 1.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 1.0, 1.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 1.0, 1.0]],
+                    [[-1.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, -1.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, -1.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 0.0, -1.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 0.0, -1.0]],
+                    [[2.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 2.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 2.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 2.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 0.0, 2.0]],
+                ]
+            )
+        )
+    )
+    batches = inputs.shape[0]
+
+    expected_outputs = nn(inputs)
+
+    # vivado_build_binfile(vhdl_dir, binfile_dir)
+
+    serial_con = serial.Serial("/dev/tty.usbmodem1101")
+    atexit.register(exit_handler, serial_con)
```

Comment:
is there a reason why this is not using the (usually) safer, more idiomatic `with` statement?

e.g.,

```python
with serial.Serial("dev/tty.usbmodem1101") as serial_con:
    code_using_serial_con(serial_con)
    
code_that_wants_serial_con_safely_close()
```

Reply:
I don't think there is a reason. Just forgot about it. I changed it accordingly.